lambda/matching.ml

```diff
@@ -2752,57 +2752,42 @@ let mk_failaction_neg partial ctx def =
     )
   | Total -> (None, Jumps.empty)
 
-(* In line with the article and simpler than before *)
+(* In [mk_failaction_pos partial seen ctx defs],
+   - [partial] is Total if we know the current switch
```

Comment:
In `mk_failaction_neg`, we use `partial` to identify that we can just return `None, Jumps.empty` and don't bother computing fail patterns. Do you know why we don't do that short-circuiting in `mk_failaction_pos` as well in the case that `fail_pats < match_context_rows`? If we trust `Total`, then it seems that short-circuiting would save us some needless work and keep contexts a bit smaller.

Reply:
I don't know. It would also be possible to put this logic in the `if sig_complete then` check in the caller of `mk_failaction_pos` (`if sig_complete || partial = Total then ...`). (Something similar is done in `combine_variant`.) I would need to ask Luc if he considered this; and probably will. This could lead to a small code improvement in some cases of matching on an instance of a GADT type, where some cases can be eliminated by typing.